src/nevermined/Assets.ts

```diff
@@ -266,7 +329,25 @@ export class Assets extends Instantiable {
             this.logger.log('Files encrypted')
             observer.next(CreateProgressStep.FilesEncrypted)
 
+            const ddoVersion: NvmConfigVersions = {
+                id: 0,
+                updated: ddo.created,
+                checksum: ddo.checksum(ddo.shortId()),
+                immutableUrl: ''
+            }
+            ddo._nvm.versions.push(ddoVersion)
+
+            if (publishMetadata != PublishMetadata.JustMarketplaceAPI) {
+                observer.next(CreateProgressStep.DdoStoredImmutable)
+                ;({url: ddoVersion.immutableUrl, backend: ddoVersion.immutableBackend } = 
```

Comment:
I don't understand this code. Is this assigning the method return values to `ddoVersion`?